BPA/Ch.06/Mtbh.stan

```diff
@@ -23,27 +23,25 @@ parameters {
   real<lower=0,upper=3> sigma;
   // In case a weakly informative prior is used
   //  real<lower=0> sigma;
-  real<lower=-16,upper=16> eps[M];
+  vector[M] sigma_raw;
 }
 
 transformed parameters {
+  vector[M] eps;
   real alpha[T];
-  vector<lower=0,upper=1>[T] p[M];
+  vector[T] logit_p[M];
 
+  eps = sigma * sigma_raw;
   for (j in 1:T)
     alpha[j] = logit(mean_p[j]); // Define logit
   for (i in 1:M) {
-    real logit_p;
 
     // First occasion: no term for recapture (gamma)
-    logit_p = alpha[1] + eps[i];
-    p[i][1] = inv_logit(logit_p);
+    logit_p[i, 1] = alpha[1] + eps[i];
 
     // All subsequent occasions: includes recapture term (gamma)
-    for (j in 2:T) {
-      logit_p = alpha[j] + eps[i] + gamma * y[i, j - 1];
-      p[i][j] = inv_logit(logit_p);
-    }
+    for (j in 2:T)
+      logit_p[i, j] = alpha[j] + eps[i] + gamma * y[i, j - 1];
```

Comment:
If you only use `y` as a predictor, as in the definition of `logit_p`, then you can declare it as a `matrix`.  You can assign integers to matrices in data.